### PR TITLE
HGCal ID variables

### PR DESCRIPTION
**Pull request by SohamBhattacharya**
#### PR description:

This is a backport of PR #32519.

### CODE REVIEW

**Comment by fwyzard** [comment]
please test

**Comment by fwyzard** [comment]
@SohamBhattacharya is there a corresponding PR for master / 11.3.x ?

**Comment by fwyzard** [comment]
code-checks

**Comment by SohamBhattacharya** [comment]
> @SohamBhattacharya is there a corresponding PR for master / 11.3.x ?

@fwyzard Yes, I have just submitted the PRs for 11_3_x (#32519) and 11_2_X (#32520).

**Comment by silviodonato** [comment]
urgent
for the HLT TDR release (11_1_X)

**Comment by silviodonato** [comment]
please test

**Comment by silviodonato** [comment]
please abort

**Comment by qliphy** [comment]
please test

**Comment by qliphy** [comment]
please test

**Comment by jpata** [comment]
hi, would you mind bringing this up to date wrt. the master PR #32519?

**Comment by SohamBhattacharya** [comment]
> hi, would you mind bringing this up to date wrt. the master PR #32519?

Done.

**Comment by perrotta** [comment]
backport of  #32519

**Comment by perrotta** [comment]
please test

**Comment by Martin-Grunewald** [comment]
+1

File: RecoEgamma/EgammaTools/src/HGCalShowerShapeHelper.cc [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+  }
+
+  if (energyNorm <= 0.0) {
+    edm::LogWarning("HGCalShowerShapeHelper")
+        << "Encountered negative or zero energy for HGCal R-variable denominator: " << energyNorm << std::endl;
+  }
+
+  double cylinderR2 = cylinderR * cylinderR;
+
+  double Rvar = 0.0;
```
Comment (jpata) [inline RecoEgamma/EgammaTools/src/HGCalShowerShapeHelper.cc:149]: as in #32519, it should be `rVar`

**Comment by jpata** [comment]
@cmsbuild please test

@SohamBhattacharya can you confirm that the latest version, post updates, runs successfully in the HLT setup in 11_1_X?

**Comment by SohamBhattacharya** [comment]
> @cmsbuild please test
> 
> @SohamBhattacharya can you confirm that the latest version, post updates, runs successfully in the HLT setup in 11_1_X?

@jpata Yes, the latest version as of now, runs successfully in the HLT setup in 11_1_X (the test config described in #32519 runs successfully in 11_1_X).

**Comment by jpata** [comment]
+reconstruction
- backport of #32519 
- jenkins tests pass (no difference expected as HLT code does not run in jenkins)
- the code has been validated in the HLT setup https://github.com/cms-sw/cmssw/pull/32517#issuecomment-758304230
- reco confirmed independently that the test config runs in 11_1_X using https://github.com/cms-sw/cmssw/pull/32519#issuecomment-754657044

**Comment by Martin-Grunewald** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**